backend/app/Actions/Pages/GetPageViewsAction.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Pages;
+
+use App\Actions\GetByIdRequest;
+use App\Exceptions\PageNotFoundException;
+use App\Repositories\Contracts\PageRepository;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
+
+final class GetPageViewsAction
+{
+    private $pageRepository;
+
+    public function __construct(PageRepository $pageRepository)
+    {
+        $this->pageRepository = $pageRepository;
+    }
+
+    public function execute(GetByIdRequest $request): GetPageViewsResponse
+    {
+        try {
+            $page = $this->pageRepository->getPageById($request->getId());
```

Comment:
this request not for particular web page. It should retrieve all page views for chosen date range in particular period and return a collection of visits per period. So, it means, we need to choose startDate, endDate and some period (hour, day, month, year), for each period calculate count of views, for example startDate: 2019/09/05, endDate: 2019/09/07, period: 24*60*60 sec, and result [{ date: 2019/09/05, visists: 123 }, { date: 2019/09/06, visists: 64 }, { date: 2019/09/07, visists: 105 }]

Reply:
the structure of request will be: startDate, endDate and period? And I had to add calculate method due to the request data in page repository also?